zkg

```diff
@@ -456,10 +571,13 @@ def cmd_install(manager, args, config, configfile):
                         invalid_reason)
             sys.exit(1)
 
+        pkg_dependencies = generate_pkg_dependencies(manager, package_infos + new_pkgs)
+        manager.set_pkg_dependencies(pkg_dependencies)
```

Comment:
This pattern is in multiple places, but will just comment here to apply for all:

This is again going down the lines of separating the kinds of things `zkg` does and `Manager` does and if I understand the purpose of inserting this here, this is the overall picture of the `zkg` operation here:

1. Update the "dependency graph" information stored in `manager.pkg_dependencies`  with any *potential* changes that will occur as a result of the `install` operation
2. Install all new packages (including ones pulled in from dependencies) via `manager.install(...)`
3. Call `manager.load_with_dependencies(...)`, which utilizes the `manager.pkg_dependencies` info that was initially modified at step (1)

The first thing I wonder is whether the updating of `manager.pkg_dependencies` should instead happen at step (2) and within the `manager.install()`. If the dependency information in `manager.pkg_dependencies` is always meant to be closely associated with "installed packages", then the place to update `manager.pkg_dependencies` seems like it ought to be wherever the actual install/removal of a package happens and right next to where `manager.installed_pkgs` gets modified.

Second thing to wonder is if it helps at all to change `manager.installed_pkgs` to have this dependency information ready to access in a convenient form rather than try to maintain it separately in `manager.pkg_dependencies`.  Not strongly suggesting that has happen for this PR (as long as the two are sufficiently in-sync with details hidden from "user"), just wondering if you thought about whether that could help simplify anything.

Reply:
At the point where we update the dependency graph, we have knowledge of the packages to be installed (package_infos) and the new packages that will be pulled in (new_pkgs). I did not want to modify the existing manager.install function, so I kept the functionality separate for now. We can merge it if that is more convenient.

Reply:
The `manager.validate_dependencies()` call doesn't actually change any state in the `manager` itself, it's just returning an answer of whether a proposed addition to the dependency graph has a solution and what that solution is.

So thinking it is best to wait to update manager's view of installed packages and their dependencies until the actual `install` operation takes place.  Can you take a look at whether things still work out doing it that way?  I think doing the update at this one location could more clearly indicate that the two states are being kept in-sync:

https://github.com/zeek/package-manager/blob/6a2bbb3e705ffaa7fe9c5c3dabc288d85122dd97/zeekpkg/manager.py#L2298-L2300

Reply:
Okay. Just to confirm again, I should investigate if I can defer the building of the dependencies until the actual installation has been done, right?

Reply:
Fixed.